worker/manager/src/mirrors_qa_manager/worker.py

```diff
@@ -291,16 +292,29 @@ def update_countries_list(self):
             "countries."
         )
 
-    def fetch_tests(self) -> list[dict[str, Any]]:
+    def fetch_tests(self) -> Iterable[dict[str, str]]:
         logger.debug("Fetching tasks from backend API")
-
         # Fetch tasks that were assigned to the worker that haven't been expired
         params = urlencode({"worker_id": self.worker_id, "status": "PENDING"})
-        data = self.query_api(f"/tests?{params}")
+        while True:
+            data = self.query_api(f"/tests?{params}")
+            nb_tests = data["metadata"]["page_size"]
+            if nb_tests == 0:  # No more pending tests to fetch
+                raise StopIteration
+
+            logger.info(f"Fetched {nb_tests} test(s) from Backend API")
+
+            current_page = data["metadata"]["current_page"]
+            last_page = data["metadata"]["last_page"]
+
+            logger.debug(
+                f"Fetched page {current_page} of {last_page} of pending tests."
+            )
 
-        logger.info(f"Fetched {data['metadata']['page_size']} test(s) from Backend API")
+            if current_page == last_page:
+                raise StopIteration
```

Comment:
Wont this prevent the yield stmt to be called for the last page?

Reply:
Oh, wow. Thanks for pointing that out.